src/local-history/local-history-manager.ts

```diff
@@ -332,4 +337,36 @@ export class LocalHistoryManager {
             vscode.commands.executeCommand('undo');
         }
     }
+
+    /**
+     * Clears all history files which are last modified @param days ago.
+     * @param days The number of days.
+     */
+    public removeOldFiles(days: number): void {
+        const dirPath = path.join(os.homedir(), LOCAL_HISTORY_DIRNAME);
+        if (!fs.existsSync(dirPath)) {
+            return;
+        }
+        try {
+            const folders = fs.readdirSync(dirPath);
+            const currentDate = Date.now();
+            let counter: number = 0;
+            for (const folder of folders) {
+                const hashedFolderPath = path.join(os.homedir(), LOCAL_HISTORY_DIRNAME, folder);
+                const files = fs.readdirSync(hashedFolderPath);
+                files.filter((file) => {
+                    const filePath = path.join(hashedFolderPath, file);
+                    const mTime = fs.statSync(filePath).mtimeMs;
+                    if (currentDate - mTime > days * DAY_TO_MILLISECONDS) {
+                        fs.unlinkSync(filePath);
+                        counter++;
+                    }
+                });
+            }
+            vscode.window.showInformationMessage(counter > 0 ? `Successfully deleted ${counter} local-history file(s)` : `No local-history found older than ${days} day(s)`);
```

Comment:
[minor] We should also include similar checks for `counter`. `counter > 1 ?  add s after history files : ''`. Also, do you think we should change the message to `Successfully deleted x old history files`?

Reply:
@kaiyue0329 I think the idea was to keep the messages simple, is there a reason for adding the additional check?

Reply:
@vince-fugnitto I missed the `(s)` in `local-history file(s)`. I think keeping the message simple is good.